### ISSUE TITLE
useCMakeServer option no longer respected

### ISSUE THREAD

**Issue by OXINARF** [microsoft/vscode-cmake-tools#1059]
### Brief Issue Summary

The `useCMakeServer` option is no longer respected, although it is still a valid option.

### Expected:

Either the option is removed (and, if currently set to false, it is transitioned to "legacy" in the new `cmakeCommunicationMode` option) or the option is respected.

### Platform and Versions

- **Operating System**: Windows, with SSH extension on Linux
- **CMake Version**: 3.10.3
- **VSCode Version**: 1.42.0
- **CMake Tools Extension Version**: 1.3.0
- **Compiler/Toolchain**: GCC 7.4.0

**Comment by andreeis** [comment]
We deprecated the useCMakeServer setting in release 1.3.0 and we are going to remove it completely in 1.4. 
However, it should still work. What is the scenario when it is not respected?

**Comment by OXINARF** [comment]
@andreeis Sorry for the delay in getting back to you.

> We deprecated the useCMakeServer setting in release 1.3.0

Is that documented anywhere? I couldn't find it but maybe I missed it.

> However, it should still work. What is the scenario when it is not respected?

I had `useCMakeServer` set to `false` and that's not respected anymore.

Looking at the change that introduced File API (https://github.com/microsoft/vscode-cmake-tools/commit/0b55e5bea912bcdb50920525ce4f35954bd31132), it shows that "conversion" is only done when `useCMakeServer` is set to `true`, which doesn't make much sense since that was the default before 1.3.0...

**Comment by bobbrow** [comment]
Yes, it was an oversight.  We'd be interested to understand why you prefer legacy mode over server and file api modes though.

**Comment by OXINARF** [comment]
@bobbrow Not exactly my choice. I'm using a SDK and the included CMake doesn't support file API and server mode gives an error (I couldn't find out exactly why).

**Comment by bobbrow** [comment]
Is the SDK available online, or is it proprietary?

**Comment by OXINARF** [comment]
Unfortunately it's proprietary. I'm not sure it is CMake Tools fault that server mode doesn't work, but I couldn't find a log that said what command it is trying to run - is it written somewhere?

**Comment by bobbrow** [comment]
You can enable trace logging with the `cmake.loggingLevel` setting.  And then logs will appear in the Output Window when you select the "CMake/Build" channel in the dropdown box.

**Comment by andreeis** [comment]
CMake Tools 1.4.0 Beta is available on GitHub.
https://github.com/microsoft/vscode-cmake-tools/releases/tag/1.4.0-beta

Please try it out and let us know if you encounter any issues.

**Comment by OXINARF** [comment]
@bobbrow Sorry for the big delay, the "new world" started and I completely forgot about this. In the meanwhile, I've been able to make server mode work (it was SDK's fault), but the `cmake.loggingLevel` wasn't useful as it doesn't display the executed commands - I've tried all debug/trace/logging options and none would output it.

@andreeis @bobbrow The fix inserted by #1218 was to remove the setting, which is good, but doesn't really fix this issue because the warning (and setting `communicationMode` accordingly) only happens when `useCMakeServer` is set to `true` - this issue is exactly about the opposite case, when it is set to false.
